### ISSUE TITLE
Getting Infinite "Connecting" screen. (502 error)

### ISSUE THREAD

**Issue by gravypod** [magne4000/quassel-webserver#119]
I've noticed that there was an issue like this before but it was closed due to inactivity. 
I'm attempting to use your app behind nginx using an SSL cert from lets encrypt.

When I attempt to connect to the server, it seems my nginx says that the connection is refused. 

Information you might need to know:

1. Both the web client and core are hosted on the same server.
2. My config for both quassel and the web client are basically default. I connected to core with the desktop client just to set it up.
3. Basically everything from your project is default. Haven't changed the config at all. 
4. 

Picture of error: http://i.imgur.com/4ont8ta.png

        server {
                listen 80;
                listen 443 ssl;
                server_name {SOMEWEBSITE};
        
                if ($scheme = http) {
                        return 301 https://$server_name$request_uri;
                }
        
                # SSL Start
                ssl_certificate /etc/letsencrypt/live/{SOMEWEBSITE}/fullchain.pem;
                ssl_certificate_key /etc/letsencrypt/live/{SOMEWEBSITE}privkey.pem;
                ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
                ssl_prefer_server_ciphers on;
                ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
                # SSL End
        
                root /var/www/{SOMEWEBSITE}/public;
        
                # LetsEncrypt
                location ~ /.well-known {
                        allow all;
                }
        
                location / {
                        proxy_pass https://127.0.0.1:64443/;
                        proxy_set_header Upgrade $http_upgrade;
                        proxy_set_header Connection "upgrade";
                        proxy_http_version 1.1;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_set_header Host $host;
                        proxy_redirect off;
                }
        
        }

If you need anything else, just @gravypod me so I see my email notification. 

Thanks for your time, this software looks great and I cant wait to start using it.

**Comment by magne4000** [comment]
@gravypod Do not use proxy_pass with `https` but with `http`:
```
proxy_pass http://127.0.0.1:64080/;
```
and run the webserver with `-m http` option (it runs on port 64080 by default in http mode)

**Comment by gravypod** [comment]
I'll give thisba fry when I get out of class. Thanks.

**Comment by gravypod** [comment]
I've updated my config but I still get to the same point.

When I run this command:

> netstat -a | grep "64080"
tcp6       0      0 [::]:64080              [::]:*                  LISTEN

> service quasselweb status
 * quasselweb is running


I can verify that quassel-web is on the correct port and running. 



I can verify that

**Comment by gravypod** [comment]
I can also verify that the Nginx install is seeing the browser connecting:

128.235.8.21 - - [15/Feb/2016:14:25:30 -0500] "POST /p/connect HTTP/1.1" 502 160 "https://{MY SITE}/" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.109 Safari/537.36"

**Comment by magne4000** [comment]
It appears that quasselwebserver do not listen on IPv4 for some reason. Try this proxy_pass in nginx:
```
proxy_pass http://localhost:64080/;
```
or
```
proxy_pass http://::1:64080/;
```

**Comment by gravypod** [comment]
I attempted localhost, still no-go @magne4000.

::1 gives an error and will not load the config

2016/02/15 23:50:24 [emerg] 3160#0: invalid port in upstream "::1:64080/"

**Comment by magne4000** [comment]
What nodejs version are you on ? And also, does you server/machine possesses IPv4 address or only IPv6 ?

**Comment by gravypod** [comment]
I have tried [::1] and still no tart. 

There is a loopback on ::1 because ip addr told me

    inet6 ::1/128 scope host


The config successfully loaded with the new IPv6 line but has the same error.

The following is the output from "n", I have tried every version of node in this list:

     node/0.11.13
     node/0.12.10
     node/4.3.0
     node/5.6.0

**Comment by magne4000** [comment]
Can you paste some nginx logs that happens during the error ?

**Comment by gravypod** [comment]
@magne4000 I didn't want to mess around with getting the formatting correct in this comment section: 

http://pastebin.com/Sgw6dFKg

**Comment by magne4000** [comment]
Okay, so not much information, but I'm pretty sure now that nothing can be done on nginx side.
Does nodejs print some error message ? Also, could you try to connect directly to quassel-webserver, and not through nginx, in order to see if we have a different behavior.

**Comment by gravypod** [comment]
@magne4000 It looks like I have the infinite loop again.

**Comment by gravypod** [comment]
Or rather to say that it still says it is connecting.

**Comment by magne4000** [comment]
If you have the same issue without nginx I suggest to check your quassel-core logs. It also can be some firewall issue (but here I will be of no help).
And if you have any nodejs error or weird logs in your terminal, do not hesitate to share them.

**Comment by gravypod** [comment]
@magne4000 this is very strange. This is from quassel-web's console output.

http://pastebin.com/bk8ddurD

**Comment by magne4000** [comment]
Is this through nginx or with direct connection ?

**Comment by gravypod** [comment]
@magne4000 yes

**Comment by magne4000** [comment]
Is this possible to have the same paste but regarding a direct connection ?

**Comment by gravypod** [comment]
@magne4000 http://pastebin.com/6xSMDfdg

**Comment by magne4000** [comment]
So yup, I suggest to look at your quasselcore logs, you should have some useful information in it (I hope)

**Comment by gravypod** [comment]
@magne4000 where would I find those....

**Comment by gravypod** [comment]
Sorry, I am new to this entire ecosystem

**Comment by magne4000** [comment]
It depends on you OS, but under Ubuntu for example it's under `/var/log/quassel/core.log`

**Comment by magne4000** [comment]
If you do not have any info in this file, it means that your quassel core probably doesn't listen on `127.0.0.1:4242`, perhaps it only listens on [::1]:4242 or on another port. Perhaps try to use `localhost` instead of `127.0.0.1` in the web interface

**Comment by gravypod** [comment]
@magne4000 I just noticed something strange. My quasselcore service seems to crash as soon as the quassel web tries to connect to it.

When I connect with my client, it opens the configuration window again (for core) but it crashes as soon as I'm done with configuration.

**Comment by magne4000** [comment]
So you have a core issue it seems. First resolve this issue (perhaps install a newer version for sources), and then try again.

**Comment by gravypod** [comment]
@magne4000 yep, trying that now. I'll keep you posted. I have to head to class soon, so it might be a little bit. You've been a great help so far!

**Comment by magne4000** [comment]
Happy to help ;)

**Comment by gravypod** [comment]
@magne4000 I found the issue!

Quassel was starting fine. When run from the user I created for the process it was fine. The issue came from the init script that I found for quasselcore. It used /var/cache/quassel as the datadir for the configs. 

Since there was no folder, it seems that when quassel went to write the configs it just died!

Sorry to bug you with this, I just wanted to leave this here just in case someone else has the same issue.

Consider this issue closed.
